packages/text-input/src/TextInput.tsx

```diff
@@ -295,9 +295,11 @@ const TextInput = React.forwardRef(
 
     return (
       <div className={cx(textInputStyle, className)}>
-        <label htmlFor={id} className={labelStyle}>
-          {label}
-        </label>
+        {label && (
```

Comment:
Hmm.... I wonder if there's a way to enforce there being some label for the input. Maybe require `aria-labelledby` if label isn't present?

Reply:
Great call -- updated this.